src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -208,14 +208,40 @@ services:
     arguments:
       $countryChoices: '@=service("prestashop.core.form.choice_provider.country_by_iso_code").getChoices()'
 
-  PrestaShopBundle\Form\Admin\Improve\Payment\Preferences\PaymentModulePreferencesType:
+  PrestaShopBundle\Form\Admin\Improve\Payment\Preferences\AbstractPaymentModuleRestrictionsType:
+    arguments:
+      $paymentModules: '@=service("prestashop.adapter.module.payment_module_provider").getPaymentModuleList()'
+    tags:
+      - { name: form.type }
+
+  PrestaShopBundle\Form\Admin\Improve\Payment\Preferences\PaymentModuleCarrierRestrictionsType:
     arguments:
       $paymentModules: '@=service("prestashop.adapter.module.payment_module_provider").getPaymentModuleList()'
-      $countryChoices: '@=service("prestashop.core.form.choice_provider.country_by_id").getChoices()'
-      $groupChoices: '@=service("prestashop.core.form.choice_provider.group_by_id").getChoices()'
       $carrierChoices: '@=service("prestashop.core.form.choice_provider.carrier_by_reference_id").getChoices()'
-      $currencyChoices: '@=service("prestashop.core.form.choice_provider.currency_by_id").getChoices()'
+    tags:
+      - { name: form.type }
+
+  PrestaShopBundle\Form\Admin\Improve\Payment\Preferences\PaymentModuleCountryRestrictionsType:
+    arguments:
+      $paymentModules: '@=service("prestashop.adapter.module.payment_module_provider").getPaymentModuleList()'
+      $countryChoices: '@=service("prestashop.core.form.choice_provider.country_by_id").getChoices()'
       $countryDataProvider: '@prestashop.adapter.data_provider.country'
+    tags:
+      - { name: form.type }
+
+  PrestaShopBundle\Form\Admin\Improve\Payment\Preferences\PaymentModuleCurrencyRestrictionsType:
+    arguments:
+      $paymentModules: '@=service("prestashop.adapter.module.payment_module_provider").getPaymentModuleList()'
+      $currencyChoices: '@=service("prestashop.core.form.choice_provider.currency_by_id").getChoices()'
```

Comment:
don't call get choices, just inject CurrencyByIdChoiceProvider, same for $paymentModules. And it should be autowired